tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp

```diff
@@ -0,0 +1,209 @@
+    vector<_Chrono_specs<CharT>>& expected_chrono_specs;
+    size_t curr_index = 0;
+
+    constexpr void _On_align(_Align aln) {
+        assert(aln == expected_alignment);
+    }
+    constexpr void _On_fill(basic_string_view<CharT> str_view) {
+        assert(str_view == expected_fill);
+    }
+    constexpr void _On_width(int width) {
+        assert(width == expected_width);
+    }
+    constexpr void _On_dynamic_width(size_t id) {
+        assert(id == expected_dynamic_width);
+    }
+    constexpr void _On_dynamic_width(_Auto_id_tag) {
+        assert(expected_auto_dynamic_width);
+    }
+    constexpr void _On_precision(int pre) {
+        assert(pre == expected_precision);
+    }
+    constexpr void _On_dynamic_precision(size_t id) {
+        assert(id == expected_dynamic_precision);
+    }
+    constexpr void _On_dynamic_precision(_Auto_id_tag) {
+        assert(expected_auto_dynamic_precision);
+    }
+    constexpr void _On_conversion_spec(CharT mod, CharT type) {
+        char _Char_mod  = static_cast<char>(expected_chrono_specs[curr_index]._Modifier);
+        char _Char_type = static_cast<char>(expected_chrono_specs[curr_index]._Type);
+
+        assert(mod == _Char_mod);
+        assert(type == _Char_type);
+        assert(expected_chrono_specs[curr_index]._Lit_char == CharT{0}); // not set
+        ++curr_index;
+    }
+    constexpr void _On_lit_char(CharT ch) {
+        assert(ch == expected_chrono_specs[curr_index]._Lit_char);
+        assert(expected_chrono_specs[curr_index]._Modifier == '\0'); // not set
+        assert(expected_chrono_specs[curr_index]._Type == '\0'); // not set
+        ++curr_index;
+    }
+};
+
+template <typename CharT, typename callback_type>
+constexpr void test_parse_helper(const CharT* (*func)(const CharT*, const CharT*, callback_type&&),
+    basic_string_view<CharT> view, bool err_expected = false,
+    typename basic_string_view<CharT>::size_type expected_end_position = basic_string_view<CharT>::npos,
+    callback_type&& callbacks                                          = {}) {
+    try {
+        auto end = func(view.data(), view.data() + view.size(), move(callbacks));
+        if (expected_end_position != basic_string_view<CharT>::npos) {
+            assert(end == view.data() + expected_end_position);
+        }
+        assert(!err_expected);
+    } catch (const format_error&) {
+        assert(err_expected);
+    }
+}
+
+template <typename CharT>
+constexpr bool test_parse_conversion_spec() {
+    auto parse_conv_spec_fn = _Parse_conversion_specs<CharT, testing_callbacks<CharT>>;
+    using view_typ          = basic_string_view<CharT>;
+    using chrono_spec       = _Chrono_specs<CharT>;
+
+    view_typ s0(TYPED_LITERAL(CharT, "%B"));
+    view_typ s1(TYPED_LITERAL(CharT, "%Ec"));
+    view_typ s2(TYPED_LITERAL(CharT, "%Od"));
+    view_typ s3(TYPED_LITERAL(CharT, "%E"));
+    view_typ s4(TYPED_LITERAL(CharT, "%"));
+    view_typ s5(TYPED_LITERAL(CharT, "%}"));

```

Comment:
Should we also have test coverage for `"%E}"`? I think that will cover all of the codepaths of concern.